tools/gen-user-data

```diff
@@ -48,27 +46,26 @@ def main():
     p.add_argument("configdir", metavar="CONFIGDIR", help="input directory")
     args = p.parse_args()
 
-    try:
-        with open(f"{args.configdir}/user-data.yml") as f:
-            userdata = yaml.load(f, Loader=yaml.SafeLoader)
-    except FileNotFoundError:
-        userdata = {}
+    write_files = []
 
     filesdir = f"{args.configdir}/files"
     for directory, dirs, files in os.walk(filesdir, followlinks=True):
         for name in files:
             path = f"{directory}/{name}"
             with open(path, "rb") as f:
                 content = base64.b64encode(f.read()).decode("utf-8")
-            userdata.setdefault("write_files", []).append({
+            write_files.append({
                 "path": "/" + os.path.relpath(path, filesdir),
                 "encoding": "b64",
                 "content": content,
                 "permissions": octal_mode_string(os.lstat(path).st_mode)
             })
 
-    sys.stdout.write("#cloud-config\n")
-    yaml.dump(userdata, sys.stdout, Dumper=yaml.SafeDumper)
+    with open(f"{args.configdir}/user-data.yml") as f:
+        sys.stdout.write(f.read())
+    sys.stdout.write("write_files: ")
+    json.dump(write_files, sys.stdout)
```

Comment:
Do I read this correctly that an JSON oneliner is a valid array in YAML?

Reply:
Yes, as of YAML 1.2, it is a superset of JSON. The [spec](https://yaml.org/spec/1.2/spec.html) states that the "primary objective of this revision is to bring YAML into compliance with JSON as an official subset".